src/garage/tf/samplers/ray_batched_sampler.py

```diff
@@ -0,0 +1,71 @@
+"""Ray Sampler, for tensorflow algorithms.
+
+Currently the same as garage.samplers.RaySampler but includes
+support for Tensorflow sessions
+"""
+import ray
+import tensorflow as tf
+
+from garage.sampler import RaySampler, SamplerWorker
+
+
+class RaySamplerTF(RaySampler):
+    """Ray Sampler, for tensorflow algorithms.
+
+    Currently the same as garage.samplers.RaySampler
+
+    Args:
+        - Same as garage.samplers.RaySampler
+    """
+
+    def __init__(self,
+                 algo,
+                 env,
+                 seed,
+                 should_render=False,
+                 num_processors=None):
+        super().__init__(
+            algo,
+            env,
+            seed,
+            should_render=False,
+            num_processors=None,
+            sampler_worker_cls=SamplerWorkerTF)
+
+    def shutdown(self):
+        """Shuts down the worker."""
+        temp = []
+        for worker in self._all_workers.values():
```

Comment:
is there a reason this isn't simply super().shutdown()?

Reply:
So the decision here was, I manually enter a tensorflow session globally for all of my worker processes, on each one of their processes, as opposed to using a with statement to enter a tensorflow session. Does it matter if after I reap these processes, I exit all of the tensorflow sessions that I entered. If not, then I can get rid of this.

Reply:
additionally, ray itself has a shutdown routine that needs to be called (line 41)